WalletWasabi/Blockchain/BlockFilters/FilterLinkedList.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using WalletWasabi.Backend.Models;
+
+namespace WalletWasabi.Blockchain.BlockFilters;
+
+public class FilterLinkedList : IEnumerable<FilterModel>
```

Comment:
No, it doesn't because the enumerator works as all the other enumerator by checking the version of the underlying collection an throwing in case it change. That okay in most cases but filters don't change and even in case of a reorg the head could be different but all the rest not.